### DISCUSSION TITLE
How add some markers to map over intent API?

### DISCUSSION THREAD

**Discussion by dimio** [osmandapp/OsmAnd#11233] [Q&A]
Hello! I am trying to add multiple markers to a map using Intent API.
```
val markers: List<Triple<Double, Double, String>> = listOf(
            Triple(51.650143, 39.163409, "Point 1"),
            Triple(51.705979, 39.208510, "Point 2"),
            Triple(51.705980, 39.208511, "Point 3"),
            Triple(51.705982, 39.208512, "Point 4"),
        )
markers.forEach { osmHelper.addMapMarker(it.first, it.second, it.third) }
```
But only the last marker from the list ("Point 4") is added. What am I doing wrong?

**Comment by vshcherb** [comment]
It is only 1 map marker could be displayed on top, all others should be available in the list Map Markers

**Reply by dimio** [reply]
The list of markers also displays only the last one.
Adding a delay when creating markers helped.
```
markers.forEach {
    osmHelper.addMapMarker(it.first, it.second, it.third)
    Thread.sleep(500)
}
```
This is a bad solution, but it worked.

**Reply by vshcherb** [reply]
it's quite strange, that AIDL communication is "broken", the reason could be if it goes multithread somehow but I wasn't aware of this.
